packages/next/server/config-shared.ts

```diff
@@ -355,11 +355,14 @@ export interface NextConfig extends Record<string, any> {
   /**
    * By default, Next.js will automatically inline font CSS at build time
    *
-   * @default true
+   * @default { inlineFonts: true }
    * @since version 10.2
    * @see [Font Optimization](https://nextjs.org/docs/basic-features/font-optimization)
    */
-  optimizeFonts?: boolean
+  optimizeFonts?: {
```

Comment:
```suggestion
  optimizeFonts?: boolean | {
```
Similar to https://github.com/vercel/next.js/pull/40185#discussion_r970181210 we also need to allow this type here as well and then we can massage the boolean value into the object variant in `server/config.ts`

Reply:
Done